components/capture-build-scan-url-maven-extension/src/main/java/com/gradle/RootProjectExtractor.java

```diff
@@ -1,73 +0,0 @@
-package com.gradle;
-
-import org.apache.maven.execution.ExecutionEvent;
-import org.apache.maven.execution.MavenExecutionRequest;
-import org.apache.maven.execution.MavenSession;
-import org.apache.maven.model.building.ModelProcessor;
-import org.apache.maven.project.MavenProject;
-import org.apache.maven.project.ProjectBuilder;
-import org.apache.maven.project.ProjectBuildingException;
-import org.codehaus.plexus.logging.Logger;
-
-import javax.inject.Inject;
-import java.io.File;
-import java.util.*;
-
-import static java.util.Collections.emptyList;
-
-public class RootProjectExtractor {
-    private final ProjectBuilder projectBuilder;
-    private final ModelProcessor modelProcessor;
-    private final Logger logger;
-
-    @Inject
-    public RootProjectExtractor(ProjectBuilder projectBuilder, ModelProcessor modelProcessor, Logger logger) {
-        this.projectBuilder = projectBuilder;
-        this.modelProcessor = modelProcessor;
-        this.logger = logger;
-    }
-
-    public MavenProject extractRootProject(ExecutionEvent event) {
-        return extractRootProject(event.getSession());
-    }
-
-    public MavenProject extractRootProject(MavenSession session) {
-        List<MavenProject> allProjects = discoverAllProjects(session.getAllProjects());
-        File workspaceDirectory = getWorkspaceDirectory(session);
-
-        if (workspaceDirectory.equals(allProjects.get(0).getBasedir())) {
-            return allProjects.get(0);
-        }
-
-        File workspaceDirectoryPom = modelProcessor.locatePom(workspaceDirectory);
-        if (workspaceDirectoryPom.exists()) {
-            try {
-                return projectBuilder.build(workspaceDirectoryPom, session.getProjectBuildingRequest()).getProject();
-            } catch (ProjectBuildingException e) {
-                logger.error("Exception locating the top level project", e);
-            }
-        }
-
-        // We didn't successfully identify the root project, so just return the first project.
-        return allProjects.get(0);
-    }
-
-    private File getWorkspaceDirectory(MavenSession session) {
-        MavenExecutionRequest request = session.getRequest();
-        try {
-            return request.getMultiModuleProjectDirectory();
-        } catch (NoSuchMethodError ignored) {
-            return new File(session.getExecutionRootDirectory());
-        }
-    }
-
-    /**
-     * Older Maven versions under-reported the list of discovered projects.
```

Comment:
I think this comment is relevant for why this piece of code exists. Have you checked the output with older versions of Maven? It seems this is already a no-op in newer maven versions although this function does not seem this way.

Reply:
This is a really good question. Nice catch. I only tried with `3.8.x` which is relatively new.

The idea with this change is that we will only determine the root project name from a Build Scan, rather than from the project itself. Therefore, as long as the Gradle Enterprise API returns a project name, the build validation scripts will display it. I do believe it is possible to create a Build Scan without a project name, but I'm not sure we want to support that case as it's always recommended to set one.